Source/engine.cpp

```diff
@@ -157,7 +167,41 @@ void CelBlitSafeTo(CelOutputBuffer out, int sx, int sy, BYTE *pRLEBytes, int nDa
 	}
 }
 
-void CelClippedDrawSafeTo(CelOutputBuffer out, int sx, int sy, BYTE *pCelBuff, int nCel, int nWidth)
+void CelBlitSafeTo_CropY(CelOutputBuffer out, int sx, int sy, BYTE *pRLEBytes, int nDataSize, int nWidth, int startY, int endY)

```

Comment:
Maybe instead of yet another blitter routine, we could limit the target buffer like in minitext.cpp: https://github.com/diasurgical/devilutionX/blob/master/Source/minitext.cpp#L147

Reply:
Done.